src/Amazon.Lambda.Tools/Commands/UpdateFunctionConfigCommand.cs

```diff
@@ -703,6 +709,20 @@ private UpdateFunctionConfigurationRequest CreateConfigurationRequestIfDifferent
                 }
             }
 
+            var snapStartApplyOn = this.GetStringValueOrDefault(this.SnapStartApplyOn, LambdaDefinedCommandOptions.ARGUMENT_SNAP_START_APPLY_ON, false);
+            if (!string.IsNullOrEmpty(snapStartApplyOn))
+            {
+                if (null == request.SnapStart)
+                    request.SnapStart = new SnapStart();
+
+                if (!string.Equals(existingConfiguration?.SnapStart?.ApplyOn?.Value, snapStartApplyOn, StringComparison.Ordinal))
+                {
+                    request.SnapStart.ApplyOn = snapStartApplyOn;
```

Comment:
Can request.SnapStart be null here based on the if check?

Reply:
`existingConfiguration.SnapStart` is right. We are checking to see if the specified by the user value `snapStartApplyOn` is different what is current set on the function ``existingConfiguration.SnapStart`. If they are then set the new value on the patch update `request` object.